module-system/sov-modules-stf-template/src/app_template.rs

```diff
@@ -23,10 +24,11 @@ use sov_zk_cycle_macros::cycle_tracker;
 /// An implementation of the
 /// [`StateTransitionFunction`](sov_rollup_interface::stf::StateTransitionFunction)
 /// that is specifically designed to work with the module-system.
-pub struct AppTemplate<C: Context, Da: DaSpec, Vm, RT: Runtime<C, Da>> {
+pub struct AppTemplate<C: Context, Da: DaSpec, Vm, RT: Runtime<C, Da>, K: Kernel<C, Da>> {
     /// State storage used by the rollup.
     /// The runtime includes all the modules that the rollup supports.
     pub(crate) runtime: RT,
+    pub(crate) kernel: K,
```

Comment:
Question: Will we have a finite set of `Kernel` implementations or do we expect the users co create their own `Kernels`? If we have a finite set we could have something like:
```
enum Kernels{
 BasicKernel(BC),
 SomeOtherKernel(..),
...
}
```
And choose the correct one from the runtime config, this way we could avoid introducing the additional generic.

Reply:
I think we will probably be able to reduce this to a finite set, but I'm not quite sure yet